### PR TITLE
Expand options documentation test and examples for Ruby

### PR DESCRIPTION
**Pull request by aguspe**
### Description
This PR adds extra tests for the options in ruby and updates the documentation examples, besides that, it also moves the proxy example from the documentation file

### Motivation and Context
For new and experienced user to be able to fully understand all the options offered by Selenium is important to have up-to-date documentation with examples

### Types of changes
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

## **Type**
Enhancement, Tests, Documentation

___

## **Description**
- Added multiple new tests in Ruby for Selenium WebDriver options including handling untrusted certificates, unhandled prompts, window rect, strict file interactability, and proxy settings.
- Updated English, Japanese, Brazilian Portuguese, and Chinese WebDriver documentation to link to the new Ruby code examples.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options_spec.rb</strong><dd><code>Expand Selenium WebDriver Options Tests in Ruby</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/ruby/spec/drivers/options_spec.rb

<li>Added tests for handling untrusted certificates, unhandled prompts, <br>window rect, strict file interactability, and proxy settings in Chrome <br>and Firefox.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-14ec885660d66546724e448a9870b3afdd72ec096c7d4777539e3e4dfb65c39c">+45/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>options.en.md</strong><dd><code>Update English WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.en.md

<li>Updated Ruby code examples with links to the new tests in the <br>options_spec.rb.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-62bab9f28d27a7f462881a28eca384a0e534a730f849248f396d171820bc748e">+5/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.ja.md</strong><dd><code>Update Japanese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.ja.md

<li>Updated Japanese WebDriver documentation to include new Ruby test <br>examples.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-2ed475220bde1368cfa3883d71db179409ddaf9ccbaea847ed3f6ffe816321e4">+5/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.pt-br.md</strong><dd><code>Update Brazilian Portuguese WebDriver Options Documentation with Ruby </code><br><code>Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.pt-br.md

<li>Updated Brazilian Portuguese WebDriver documentation to include new <br>Ruby test examples.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-6eaa598a13e9d64c6afbd54b84aef470520557f08c4d183ffb28ceb9ed04d0bb">+5/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>options.zh-cn.md</strong><dd><code>Update Chinese WebDriver Options Documentation with Ruby Examples</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/drivers/options.zh-cn.md

<li>Updated Chinese WebDriver documentation to include new Ruby test <br>examples.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1679/files#diff-9b23b749355db8154bfcc4a1d70f809b7d4e91a7b3044f9ac74a6c1a005873d6">+5/-11</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by aguspe** [comment]
Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?

If it does, I can also update the timeouts example on this PR

Screenshot 2024-04-19 at 20 40 42

File: examples/ruby/spec/drivers/options_spec.rb [hunk 1]
```diff
@@ -45,5 +45,50 @@
...
+      driver.get('https://expired.badssl.com/')
+      driver.quit
+    end
+
+    it 'sets unhandled prompt behavior' do
+      options = Selenium::WebDriver::Options.chrome
+      options.unhandled_prompt_behavior = :accept
+
+      driver = Selenium::WebDriver.for :chrome, options: options
+      driver.get('https://www.google.com')
```
Comment (harsha509) [inline examples/ruby/spec/drivers/options_spec.rb]: can we use https://www.selenium.dev/selenium/web/

File: examples/ruby/spec/drivers/options_spec.rb [hunk 2]
```diff
@@ -45,5 +45,50 @@
...
+      driver.quit
+    end
+
+    it 'sets window rect' do
+      options = Selenium::WebDriver::Options.firefox
+      options.set_window_rect = true
+
+      driver = Selenium::WebDriver.for :firefox, options: options
+      driver.get('https://www.google.com')
```
Comment (harsha509) [inline examples/ruby/spec/drivers/options_spec.rb]: can we use https://www.selenium.dev/selenium/web/ for all navigations etc.

**Review by harsha509 — CHANGES_REQUESTED** [review]

**Comment by harsha509** [comment]
> Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?
> 
> If it does, I can also update the timeouts example on this PR
> 
> Screenshot 2024-04-19 at 20 40 42

Hi @aguspe,

The timeout mentioned refers to the W3C capability timeout in W3C WebDriver, see https://www.w3.org/TR/webdriver2/#capabilities

Also, It covers features like `ScriptTimeout`, `PageLoadTimeout`, and `ImplicitWaitTimeout`, as per the WebDriver documentation. see https://www.w3.org/TR/webdriver2/#timeouts

So, I believe it's correctly included in the driver options section.

Thanks,
Sri

**Comment by aguspe** [comment]
> > Hello @diemol one question about the documentation, does it make sense that the timeouts are under the options page and not on its page?
> > If it does, I can also update the timeouts example on this PR
> > Screenshot 2024-04-19 at 20 40 42
> 
> Hi @aguspe,
> 
> The timeout mentioned refers to the W3C capability timeout in W3C WebDriver, see https://www.w3.org/TR/webdriver2/#capabilities
> 
> Also, It covers features like `ScriptTimeout`, `PageLoadTimeout`, and `ImplicitWaitTimeout`, as per the WebDriver documentation. see https://www.w3.org/TR/webdriver2/#timeouts
> 
> So, I believe it's correctly included in the driver options section.
> 
> Thanks, Sri

Thank you so much for the help I have updated this PR to cover the timeout examples :)

**Review by harsha509 — APPROVED** [review]

**Comment by aguspe** [comment]
Hi @harsha509 before this gets merged I need to update some lines because I added a new let on the Rspec test so I'm missing some updates

**Comment by harsha509** [comment]
HI @aguspe ,

can you do it in a new PR if possible ? let me know

Thanks,
Sri

**Comment by aguspe** [comment]
> HI @aguspe ,
> 
> can you do it in a new PR if possible ? let me know
> 
> Thanks, Sri

I just updated it in a commit here and I fixed a documentation bug, I hope that's okey

**Comment by aguspe** [comment]
Hello @harsha509 would it be possible to merge this PR? I will create a new PR to expand the selenium docs and correct an issue with the upgrade selenium ruby example but I keep merging trunk into this PR because it gets out of sync :)

Thank you so much

**Comment by aguspe** [comment]
This PR needs to be merged before #1695

**Review by harsha509 — APPROVED** [review]

**Comment by harsha509** [comment]
Thank you @aguspe !

**Pull request merged**